File: CSharp/1. Two Sum.cs
//A +B = target
//A = target - B
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> dic = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            dic[target - nums[i]] = i;
        }

        int[] res = new int[2];
        for (int i = 0; i < nums.Length; i++)
        {
            if (dic.ContainsKey(nums[i]))
            {
                int otherIndex = dic[nums[i]];
                if (i != otherIndex)
                {
                    res[0] = Math.Min(i, otherIndex);
                    res[1] = Math.Max(i, otherIndex);
                }
            }
        }

        return res;
    }
}

File: Java/1. Two Sum.java
//A +B = target
//A = target - B
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }

        int[] res = new int[2];
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                int otherIndex = map.get(target - nums[i]);

                if (otherIndex != i) {
                    res[0] = Math.min(i, otherIndex);
                    res[1] = Math.max(i, otherIndex);
                    break;
                }
            }
        }

        return res;
    }
}